docs/reference/src/language/syntax/functions.md

```diff
@@ -14,27 +35,152 @@ Functions can be annotated with the `pure` keyword.
-    protected function set-bounds(min: int, max: int) {
-        root.min = min;
-        root.max = max
+    property <int> secret_number: my-function();
+    public pure function my-function() -> int {
+        return 1;
+    }
+
+    VerticalLayout {
+        public pure function my-function() -> int {
+            return 2;
+        }
+
+        Text {
+            text: "The secret number is " + my-function();
+            public pure function my-function() -> int {
+                return 3;
+            }
+        }
+
+        Text {
+            text: "The other secret number is " + my-function();
+        }
+    }
+}
+```
+
+In the example above, the property `secret_number` will be set to 1, and the text labels will say "The
+secret number is 3" and "The other secret number is 2".
+
+## Function visibility
+
+By default, functions are private and cannot be accessed from other components.
+
+However, their accessibility can be modified using the `public` or `protected` keywords.
+
+- A root-level function annotated with `public` can be accessed by any component.
+
+To access such a function from a different component, you always need a target, which in practice
+means the calling component must declare the called component as one of its child elements.
+
+```slint,no-preview
+export component HasFunction {
+    public pure function double(x: int) -> int {
+        return x * 2;
+    }
+}
+
+export component CallsFunction {
+    property <int> test: my-friend.double(1);
+
+    my-friend := HasFunction {
+    }
+}
+```
+
+If a function is declared in a child element, even if marked public, it is not possible to call it
+from another component, as the child elements themselves are not public and a valid target for the
+function does not exist:
+
+```slint,no-preview
+export component HasFunction {
+    t := Text {
+        public pure function double(x: int) -> int {
+            return x * 2;
+        }
     }
-    public pure function inbound(x: int) -> int {
-        return Math.min(root.max, Math.max(root.min, x));
+}
+
+export component CallsFunction {
+    // Compiler error!
+    // property <int> test: my-friend.t.double(1);
+
+    my-friend := HasFunction {
     }
 }
 ```
 
-In the example above, `set-bounds` is a protected function that updates the `min` and `max` properties of the root component.
-The `inbound` function is a public, pure function that takes an integer `x` and returns the value constrained within the `min` and `max` bounds.
+Functions marked `public` in an exported component can also be invoked from backend code (Rust, C++, JS).
+See the language-specific documentation for the generated code to use.
+
+- A function annotated with `protected` can only be accessed by components that directly inherit from it.
+
+## Functions vs. callbacks
+
+There are a lot of similarities between functions and [callbacks](callbacks.md):
+
+- They are both callable blocks of logic/code
+- They can be invoked similarly
```

Comment:
Aren't they invoked not only similarly, but in exactly the same way?

Reply:
Done -- see, as someone just learning Slint, I'm a little bit afraid of making calls like that. As far as I know, they are, but I'm not an expert, so this was just me hedging :stuck_out_tongue: